plugins/hls-overloaded-record-dot-plugin/test/testdata/Simple.hs

```diff
@@ -0,0 +1,11 @@
+data Happy = Happy {name :: String, age :: Int, happy :: Bool}
+
+main :: IO ()
+main = do
+    putStrLn test
+
+man :: Happy
+man = Happy {name = "Happy", age = 1, happy = True}
+
+test :: String
+test  = name man
```

Comment:
Other possible test cases:
```
-- multiline expression, could also try with a case
name $ do
   m <- pure man
```
```
-- check that nothing funny happens with nested dots
name (building.owner)
```
```
-- application across multiple lines
name
   man
```

Reply:
I can't see how the multiline do would make sense, as any do statement would need to be evaluated to a monadic value. Whereas something like fmap would work, currently, this plugin won't offer code hints if it's not some form of a simple application. 
However, I was able to get a multiline case, nested dots, and naive multi line working.